### PR TITLE
[BEAM-11607] Add word count tasks

### PR DESCRIPTION
**Pull request by isidroamv**
This PR adds a task to check the environment setup and tasks for wordCount example for Java, Python, and Go. 

R: @tysonjh 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>SDK</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_ULR/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/">
          Build Status
        </a>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Java11_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_Examples_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Python</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>---</th>
      <th>Java</th>
      <th>Python</th>
      <th>Go</th>
      <th>Website</th>
      <th>Whitespace</th>
      <th>Typescript</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Non-portable</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Portable</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_GoPortable_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by tysonjh** [comment]
It would be helpful to also see the documentation updates to go along with this to make sure the task name makes sense.

File: examples/java/build.gradle [hunk 1]
```diff
@@ -161,3 +161,10 @@ task execute (type:JavaExec) {
   systemProperties System.getProperties()
   args System.getProperty("exec.args", "").split()
 }
+
+task wordCount(type:JavaExec) {
```
Comment (tysonjh) [inline examples/java/build.gradle:165]: Can you add a comment here, and the other wordcount tasks added, to explain that they're used for validating the development environment for contributors?

**Comment by isidroamv** [comment]
> It would be helpful to also see the documentation updates to go along with this to make sure the task name makes sense.

This PR will update the documentation based on the work of BEAM-11607 and BEAM-11608 https://github.com/apache/beam/pull/14477/

**Comment by tysonjh** [comment]
When I ran this locally, it failed with the following error:

```

> Task :sdks:python:setupVirtualenv FAILED
created virtual environment CPython3.8.7.final.0-64 in 334ms
  creator CPython3Posix(dest=/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/usr/local/google/home/tysonjh/.local/share/virtualenv)
    added seed packages: pip==21.0.1, pkg_resources==0.0.0, setuptools==54.1.1, wheel==0.36.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
Traceback (most recent call last):
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 8, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/autocompletion.py", line 9, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/main_parser.py", line 7, in <module>
    from pip._internal.cli import cmdoptions
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/cmdoptions.py", line 20, in <module>
    from pip._vendor.packaging.utils import canonicalize_name
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_vendor/packaging/utils.py", line 9, in <module>
    from .tags import Tag, parse_tag
  File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_vendor/packaging/tags.py", line 7, in <module>
    import distutils.util
ModuleNotFoundError: No module named 'distutils.util'

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sdks:python:setupVirtualenv'.
> Process 'command 'sh'' finished with non-zero exit value 1
```

**Comment by isidroamv** [comment]
> When I ran this locally, it failed with the following error:
> 
> ```
> 
> > Task :sdks:python:setupVirtualenv FAILED
> created virtual environment CPython3.8.7.final.0-64 in 334ms
>   creator CPython3Posix(dest=/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555, clear=False, no_vcs_ignore=False, global=False)
>   seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/usr/local/google/home/tysonjh/.local/share/virtualenv)
>     added seed packages: pip==21.0.1, pkg_resources==0.0.0, setuptools==54.1.1, wheel==0.36.2
>   activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
> Traceback (most recent call last):
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/bin/pip", line 5, in <module>
>     from pip._internal.cli.main import main
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 8, in <module>
>     from pip._internal.cli.autocompletion import autocomplete
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/autocompletion.py", line 9, in <module>
>     from pip._internal.cli.main_parser import create_main_parser
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/main_parser.py", line 7, in <module>
>     from pip._internal.cli import cmdoptions
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_internal/cli/cmdoptions.py", line 20, in <module>
>     from pip._vendor.packaging.utils import canonicalize_name
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_vendor/packaging/utils.py", line 9, in <module>
>     from .tags import Tag, parse_tag
>   File "/usr/local/google/home/tysonjh/Development/beam/build/gradleenv/1922375555/lib/python3.8/site-packages/pip/_vendor/packaging/tags.py", line 7, in <module>
>     import distutils.util
> ModuleNotFoundError: No module named 'distutils.util'
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':sdks:python:setupVirtualenv'.
> > Process 'command 'sh'' finished with non-zero exit value 1
> ```

I already solved this issue. Try again @tysonjh

**Review by tysonjh — APPROVED** [review]
I'm not able to verify the python tests for some reason due to my local setup. I'm going to ask for someone else to try and merge. Otherwise LGTM.

**Comment by tysonjh** [comment]
R: @udim

**Comment by udim** [comment]
Runs for me successfully

**Comment by udim** [comment]
Run PythonLint PreCommit

**Comment by udim** [comment]
Run Python PreCommit

**Review by udim — APPROVED** [review]
LGTM, waiting for tests to pass

**Comment by pabloem** [comment]
I successfully ran this in my environment. I'll merge if that's okay : )

**Pull request merged**